### ISSUE TITLE
ASPP-based model (Perone et al. 2018)

### ISSUE THREAD

**Issue by hermancollin** [ivadomed/ivadomed#1287]
### Motivation for the feature
@valosekj showed interest in trying the method implemented in [Perone et al., 2018](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5899127/). The repo for the code is here: https://github.com/neuropoly/gmseg. It was implemented prior to the first release of ivadomed. Here is what the architecture looks like:
![model](https://raw.githubusercontent.com/neuropoly/gmseg/55aee10c80ae27800d62277230b44538abeecf24/docs/img/architecture.png)

According to the paper, the model proposed, based on _Atrous Spatial Pyramid Pooling_ (ASPP), uses less parameters than the U-Net. Atrous convolutions preserve the resolution of the input throughout the network, which was shown to improve accuracy for segmentation tasks, compared to Unet. Also, they don't use residual connections. The architecture is very straightforward (the original implementation can be found [in this notebook](https://github.com/neuropoly/gmseg/blob/master/notebooks/challenge-pytorch-opensource-model.ipynb)). It seems to me the models trained in 2018 for the paper are still being used in SCT's entry point [`sct_deepseg_gm`](https://spinalcordtoolbox.com/user_section/command-line.html#sct-deepseg-gm).

### Description of the feature
Would it be of interest to implement this architecture in models.py? It would make sense to me to maintain an implementation of this architecture, especially given the fact that it was partily (or entirely?) developed here at NeuroPoly.

Does anyone have any insight on the relevancy of this architecture?

**Comment by hermancollin** [comment]
@valosekj If we go forward with this, I'd like to help. 

We have to choose which parameters will be exposed to the users. For reference, this is what we have for the Unet:
https://github.com/ivadomed/ivadomed/blob/ea664d5b14ddb0b1c8597917c7b3e1d868686f88/ivadomed/models.py#L496-L497
Since the ASPP block includes BatchNorm and Dropout, we can keep these same parameters, except for `depth`. It seems to me the proposed architecture doesn't really allow for more depth. However, we could expose the number of branches in the pyramid (e.g. _c)_ in the figure above). This is the description of this part of the network:
> Then, six parallel modules with different rates (6/12/18/24), 1 × 1 convolution, and a global average pooling are used in parallel.

We can keep the 1 x 1 convolution and the global average pooling layers and expose the number of layers with dilation (4 in the case of the paper). For example, if the user choses 5 branches, the dilation rates would be (6/12/18/24/30).

Finally, I would get rid of `is_2d` for now (to keep it simple at first and get it up and running). 
The `n_filters` argument can stay - the number of feature maps is constant throughout all layers of the network, except at the last _d)_ part where it's doubled.

**Comment by jcohenadad** [comment]
Hey team! I suggest to hold off for a moment, I'd like to discuss this in our next ivadomed meeting
